apps/modernization-ui/src/design-system/date/criteria/date-range/DateRangeEntry.tsx

```diff
@@ -1,15 +1,57 @@
 import { DateBetweenCriteria } from 'design-system/date/entry';
+import { DateRangeEntryFields, useDateBetweenCriteria } from '../useDateBetweenCriteria';
+import { useEffect } from 'react';
+import styles from './date-range-entry.module.scss';
+import classNames from 'classnames';
+import { DatePicker } from 'design-system/date/picker';
 
 type DateRangeEntryProps = {
     id: string;
-    value: DateBetweenCriteria | undefined;
+    value: DateBetweenCriteria;
     onChange: (value: DateBetweenCriteria) => void;
+    onBlur?: (value: DateBetweenCriteria) => void;
 };
 
-export const DateRangeEntry = ({ id }: DateRangeEntryProps) => {
+export const DateRangeEntry = ({ id, value, onChange, onBlur }: DateRangeEntryProps) => {
+    const { state: rangeEntry, apply, clear } = useDateBetweenCriteria(value);
+
+    const handleOnChange = (field: DateRangeEntryFields) => (value: string | undefined) => {
+        if (value) {
+            apply(field, value);
+        } else {
+            clear(field);
+        }
+    };
+
+    useEffect(() => {
+        onChange(rangeEntry as DateBetweenCriteria);
+    }, [rangeEntry, onChange]);
+
     return (
-        <div id={id} aria-label="patient-search-date-range">
-            Date range entry
+        <div id={id} aria-label="patient-search-date-range" className={styles['date-range-entry']}>
+            <div className={classNames(styles['range-wrapper'])}>
+                <label htmlFor={'from'}>From</label>
+                <DatePicker
+                    id={`${id}-from`}
+                    name="from"
+                    aria-label="from"
```

Comment:
question: from what?  Are these aria-labels unique enough if there are more than one of these components on a page?